shared/templates/file_groupowner/oval.template

```diff
@@ -13,23 +13,12 @@
    {{% endif %}}
     </criteria>
   </definition>
-  {{%- if MISSING_FILE_PASS -%}}
-    {{# Any number of files can exist, from zero to any #}}
-    {{% set FILE_EXISTENCE = "any_exist" %}}
-  {{%- else -%}}
-    {{# All defined files must exist. When using regex, at least one file must match #}}
-    {{% set FILE_EXISTENCE = "all_exist" %}}
-  {{%- endif -%}}
-
 
   {{% for filepath in FILEPATH %}}
-  <unix:file_test check="all" check_existence="{{{ FILE_EXISTENCE }}}" comment="Testing group ownership of {{{ filepath }}}" id="test_file_groupowner{{{ FILEID }}}_{{{ loop.index0 }}}" version="1">
```

Comment:
What we can do to allow the `missing_file_pass` behavior is to keep the Jinja conditionals above, so that `FILE_EXISTENCE = 'any_exist` when `MISSING_FILE_PASS` is true.

And wrap the state `state_file_groupowner{{{ FILEID }}}_gid_{{{ FILEGID }}}_{{{ loop.index0 }}}` in Jinja condiitonals so that the state is included in the test when `MISSING_FILE_PASS`.

This way, any collected object that is not compliant will still be checked against the state, and obviously fail.
If no object is collected the rule will pass.

Reply:
This doesn't work, :sweat_smile: